gravitee-am-repository/gravitee-am-repository-mongodb/src/main/java/io/gravitee/am/repository/mongodb/management/AbstractUserRepository.java

```diff
@@ -448,4 +455,31 @@ private List<CertificateMongo> toMongoCertificates(List<Certificate> modelCertif
                     return mongoCertificate;
                 }).collect(Collectors.toList());
     }
+
+    private void createOrUpdateIndex() {
+        if (ensureIndexOnStart) {
+            getDeletableIndex()
+                    .doOnComplete(() -> {
+                        try {
+                            super.createIndex(usersCollection, new Document(FIELD_REFERENCE_TYPE, 1).append(FIELD_REFERENCE_ID, 1).append(FIELD_USERNAME, 1).append(FIELD_SOURCE, 1),
+                                    new IndexOptions().name(INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE_NAME_UNIQUE).unique(true));
+                        } catch (Exception e) {
+                            logger.error("An error has occurred while creating index {} with unique constraints", INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE, e);
```

Comment:
maybe better to log with `INDEX_REFERENCE_TYPE_REFERENCE_ID_USERNAME_SOURCE_NAME_UNIQUE`

Reply:
done